src/coordinates_photo.py

```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
+
+
+    observer_time = Time(observer_time)
+    # Compute Julian Date (JD)
+    jd = observer_time.jd
+
+    # Compute Greenwich Sidereal Time (GST) in degrees
+    gst = 100.46 + 0.985647 * jd
+    gst = gst % 360  # Ensure GST is within 0 to 360 degrees
+
+    # Convert GST to hours, minutes, and seconds
+    gst_hour = gst / 15
+    gst_min = (gst_hour % 1) * 60
+    gst_sec = (gst_min % 1) * 60
+
+    # Compute Local Sidereal Time (LST) in degrees
+    lst = gst + observing_location.lon.deg
+    lst = lst % 360  # Ensure LST is within 0 to 360 degrees
+
+    lst_hour = lst / 15
+    lst_min = (lst_hour % 1) * 60
+    lst_sec = (lst_min % 1) * 60
+
+    # Celestial coordinates (RA and Dec)
     object_ra, object_dec = object_coordinates
+    celestial_coord = SkyCoord(ra=object_ra, dec=object_dec, unit="deg")
 
-    lst = local_sidereal_time(observer_lon, observer_time_dt)
-    
-    azimuth, altitude = equatorial_to_azimuthal(object_ra, object_dec, lst, observer_lat)
+    # Convert to Zenith and Azimuth
+    alt_az = celestial_coord.transform_to(AltAz(obstime=observer_time, location=observing_location))
 
+    print(f"Azimuth: {alt_az.az.deg} deg\nZenith: {alt_az.alt.deg} deg")
+    
+    r, theta = altaz_to_polar(alt_az.alt.rad, alt_az.az.rad) 
+    print(f"r: {r} m\ntheta: {theta} rad /{math.degrees(theta)} deg \n")
 
-    x, y = polar_to_cartesian(radius, azimuth)
+    # en m
+    x,y = polar_to_cartesian(r,theta)
+    print(f"x: {x} m, y:{y} m")
 
-    # Convert Cartesian coordinates to GPS coordinates 
-    λ = math.atan2(y, x)
-    φ = math.atan2(z, math.sqrt(x**2 + y**2))
+    # en rad
+    object_lat, object_lon = cartesian_to_lon_lat(x,y)
+    object_lat, object_lon = math.degrees(object_lat) , math.degrees(object_lon)
+    # print(f"lon: {lon} rad / {math.degrees(lon)}deg, lat: {lat} rad/{math.degrees(lat)}deg")
 
-    object_lon = math.degrees(λ)
-    object_lat = math.degrees(φ)
+    # object_lat, object_lon = method1(alt_az.alt.rad,alt_az.az.rad)
 
     return object_lat, object_lon
 
+def altaz_to_polar(alt, az):
+    """
+    Submethod to convert altitude (radians), azimuth (radians) of an object to polar coordinates (r,theta)
+    Returns
+    -------
+    (r, theta) : tuple
+    """
+    alt_objet = 10000 # hypothesis: flying at 10km altitude
+    r = math.cos(alt) * alt_objet # en mètres
+    theta = az
+    return r, theta
+
+def polar_to_cartesian(r, theta):
+    """
+    Submethod to convert radius (radians), theta (radians) of an object 
+    to cartesian coordinates (x,y,z) in meters
+    Returns
+    -------
+    (x, y, z) : tuple
+    """
+    x = r * math.sin(theta) # axis: 
+    y = r * math.cos(theta)
+
+    return x,y
+
+def cartesian_to_lon_lat(x,y):
+    """
+    Submethod to convert cartesian coordinates (x,y,z) (we assume z = 10km) 
+    to longitude, latitude in radians
+    Returns
+    -------
+    (lon, lat) : tuple
+    """
+    z = 10000 # en m
```

Comment:
if possible make z an optional parameter for this function, with default value 10km